### ISSUE TITLE
images not found when rendering self contained document using intermediates dir

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rmarkdown#429]
Minimal RMarkdown document using the logo from rstudio.com: //www.rstudio.com/wp-content/uploads/2014/03/blue-250.png
```
![rstudio](blue-250.png)
```

Rendering command including console output:
```
> rmarkdown::render("self_contained.Rmd",intermediates_dir="/tmp",output_options=list(self_contained=TRUE))


processing file: self_contained.Rmd
  |.................................................................| 100%
  ordinary text without R code


output file: /tmp/self_contained.knit.md

/Applications/RStudio.app/Contents/MacOS/pandoc/pandoc /tmp/self_contained.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash-implicit_figures --output self_contained.html --smart --email-obfuscation none --self-contained --standalone --section-divs --template /Library/Frameworks/R.framework/Versions/3.2/Resources/library/rmarkdown/rmd/h/default.html --variable 'theme:bootstrap' --include-in-header /var/folders/ts/s940qvdj5vj1czr9qh07fvtw0000gn/T//Rtmp8lqgQv/rmarkdown-str12e934bd9affc.html --mathjax --variable 'mathjax-url:https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --no-highlight --variable highlightjs=/Library/Frameworks/R.framework/Versions/3.2/Resources/library/rmarkdown/rmd/h/highlight 
pandoc: Could not fetch blue-250.png
blue-250.png: openBinaryFile: does not exist (No such file or directory)
Error: pandoc document conversion failed with error 67
```

Output from 

`getwd()` is the directory containing both the RMarkdown document and the logo image.

**Comment by jjallaire** [comment]
Possibly related: #130 #285 #286 

@jmcphers Could you take a look at this?

**Comment by jmcphers** [comment]
Absolutely.

**Comment by jmcphers** [comment]
Hrm, interesting. This is pretty much exactly the scenario that `copy_resources` is supposed to fix, but `copy_resources` is designed to be used with *non*-self-contained documents, and as such it runs as a post-processor to fix up the already rendered document, whereas here we need to make the images (+ other content) resolvable by pandoc *prior* to render so that it can slurp them up during `--self-contained` processing. 

Two possible approaches:

1. Run pandoc in such a way that when rendering the intermediate `.md` file, it searches in the directory of the original `.Rmd` file for content (we used to do something like this, actually, but pandoc no longer supports `--data-dir` for this purpose -- see https://github.com/jgm/pandoc/issues/1572) 

2. In this specific situation (i.e. render a self-contained doc using an intermediate directory different from the input document's directory), perform a resource copying pass prior to render that scrapes document dependencies out of the input directory and copies them to the intermediates directory so pandoc can locate them.

**Comment by jmcphers** [comment]
Oh, one other approach suggests itself:

3. As in #2, but rather than copying resources to the intermediate directory, rewrite the intermediate .md file so that it refers to the images from the original directory. That way pandoc can still find them, but we don't incur the performance (and disk) penalty of copying them as intermediates. The main drawback to this approach is that it requires us to doctor a `.md` file, which might be risky since (unlike the `.html` we're already doctoring) we'd have to resort to some pretty loose string matching to find resource usages. 

Another consideration with (2) and (3) is that both involve actually doing a pre-render--we don't (currently) know how to parse Markdown for dependencies directly, and there may be dependencies in HTML, so we convert the whole thing to HTML (via, you guessed it, `::render`) to do resource discovery. That's a recursive render in this case, so we need to be careful we don't create a hole by which the sub-render can itself require this behavior, etc.

**Comment by aronatkins** [comment]
Something to keep in mind: Some folks in the tickets @jjallaire referenced were taking advantage of the intermediate Markdown file -- publishing it on GitHub Pages or other Markdown-only platforms. That situation might be best served by local resource references rather than ones outside the intermediate directory.

**Comment by aronatkins** [comment]
Confirmed that @jmcphers's PR resolved my issue.

**Comment by Thell** [comment]
If `clean=false` and `intermediates_dir=somelocation` should the generated file used as the pandoc argument for `--include-in-header /tmp/RtmpXXX/rmarkdown-strXXX.html` be relocated to `somelocation` as well? It isn't using 0.99.446.
